msal/src/main/java/com/microsoft/identity/client/internal/CommandParametersAdapter.java

```diff
@@ -203,6 +205,46 @@ public static SilentTokenCommandParameters createSilentTokenCommandParameters(
         return commandParameters;
     }
 
+    /**
+     * Adapter method to create DeviceCodeFlowCommandParameters from DeviceCodeFlowParameters
+     * @param configuration PCA configuration
+     * @param tokenCache token cache for storing results
+     * @param parameters deviceCodeFlowParameters
+     * @return DeviceCodeFlowCommandParameters
+     */
+    public static DeviceCodeFlowCommandParameters createDeviceCodeFlowWithClaimsCommandParameters(
+            @NonNull final PublicClientApplicationConfiguration configuration,
+            @NonNull final OAuth2TokenCache tokenCache,
+            @NonNull final DeviceCodeFlowParameters parameters) {
+
+        final String claimsRequestJson = ClaimsRequest.getJsonStringFromClaimsRequest(parameters.getClaimsRequest());
+
+        final Authority authority = configuration.getDefaultAuthority();
+
+        final AbstractAuthenticationScheme authenticationScheme = new BearerAuthenticationSchemeInternal();
+
+        final DeviceCodeFlowCommandParameters commandParameters = DeviceCodeFlowCommandParameters.builder()
+                .platformComponents(AndroidPlatformComponentsFactory.createFromContext(configuration.getAppContext()))
+                .applicationName(configuration.getAppContext().getPackageName())
+                .applicationVersion(getPackageVersion(configuration.getAppContext()))
+                .clientId(configuration.getClientId())
+                .isSharedDevice(configuration.getIsSharedDevice())
+                .redirectUri(configuration.getRedirectUri())
+                .oAuth2TokenCache(tokenCache)
+                .requiredBrokerProtocolVersion(MSAL_TO_BROKER_PROTOCOL_BROKER_DCF_WITH_CLAIMS_SUPPORT_MINIMUM_VERSION)
```

Comment:
this should come from the PCA configuration. It should not be hardcoded.

Reply:
removed hardcoding